src/tests/state/experiments/selectors.test.js

```diff
@@ -1,13 +1,204 @@
+          "name": "test experiment name"
+        }
+      ])
+    );
+  });
+
+  it("should return an empty List because all experiment start dates are before chosen startDates", () => {
+    let mockedState = fromJS({
+      experiments: {
+        items: [
+          {
+            start_date: Date.parse("2017-08-22"),
+            end_date: Date.parse("2019-01-20")
+          }
+        ]
+      }, 
+      dates: {
+        startDate: "2018-07-22",
+        endDate: ""
+      }
+    });
+    expect(getFilteredExperimentsByDate(mockedState)).toEqual(fromJS([]));
+  });
+
+  it("should return empty List because experiment start date is null", () => {
+    let mockedState = fromJS({
+      experiments: {
+        items: [
+          {
+            start_date: null,
+            end_date: Date.parse("2019-01-20")
+          }
+        ]
+      }, 
+      dates: {
+        startDate: "2018-07-22",
+        endDate: ""
+      }
+    });
+    expect(getFilteredExperimentsByDate(mockedState)).toEqual(fromJS([]));
+  });
+
+  it("should return an empty List because all experiment end dates are after chosen endDates", () => {
+    let mockedState = fromJS({
+      experiments: {
+        items: [
+          {
+            start_date: Date.parse("2017-08-22"),
+            end_date: Date.parse("2019-01-20")
+          }
+        ]
+      }, 
+      dates: {
+        startDate: "",
+        endDate: "2016-07-22"
+      }
+    });
+    expect(getFilteredExperimentsByDate(mockedState)).toEqual(fromJS([]));
+  });
+
+  it("should return empty List because experiment end date is null", () => {
+    let mockedState = fromJS({
+      experiments: {
+        items: [
+          {
+            start_date: Date.parse("2019-01-20"),
+            end_date: null
+          }
+        ]
+      }, 
+      dates: {
+        startDate: "",
+        endDate: "2018-07-22"
+      }
+    });
+    expect(getFilteredExperimentsByDate(mockedState)).toEqual(fromJS([]));
+  });
+});
+
+describe("timestamp helper functions (getStartDatepickerTimestamp, getEndDatepickerTimestamp) tests", () => {
+  it("should return the startDate timestamp when startDate is not null", () => {
+    let mockedState = fromJS({
+      dates: {
+        startDate: "2018-07-22"
+      }
+    });
+    expect(getStartDatepickerTimestamp(mockedState)).toEqual(1532217600000);
+  });
+
+  it("should return null when startDate is empty string", () => {
+    let mockedState = fromJS({
+      dates: {
+        startDate: ""
+      }
+    });
+    expect(getStartDatepickerTimestamp(mockedState)).toEqual(null);
+  });
+
+  it("should return the endDate timestamp when endDate is not null", () => {
+    let mockedState = fromJS({
+      dates: {
+        endDate: "2018-07-22"
+      }
+    });
+    expect(getEndDatepickerTimestamp(mockedState)).toEqual(1532217600000);
+  });
+
+  it("should return null when endDate is empty string", () => {
+    let mockedState = fromJS({
+      dates: {
+        endDate: ""
```

Comment:
How come it's an empty string inside the state but null when we get it back from the selector?  Why not just use the same value everywhere....

Reply:
oh good point!